### ISSUE TITLE
False negative when specializing tuple type alias with incorrect type arguments

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6903]
The following should produce an error but does not. This is part of the type system conformance test suite.

```python
from typing import TypeVar

T = TypeVar("T")

IntTupleGeneric = tuple[int, T]

IntTupleGeneric[*tuple[float, ...]]  # Type error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
